azkaban-common/src/test/java/azkaban/metric/MetricManagerTest.java

```diff
@@ -75,15 +81,31 @@ public void managerEmitterMaintenanceTest() {
   @Test
   public void managerEmitterHandlingTest() throws Exception {
     this.emitter.purgeAllData();
-    final Date from = new Date();
+    final Date from = DateTime.now().minusMinutes(1).toDate();
     this.metric.notifyManager();
 
-    Thread.sleep(2000);
+    this.emitterWrapper.countDownLatch.await(10L, TimeUnit.SECONDS);
```

Comment:
Great Test strategy to fix long wait.

Reply:
Looks like we can use the same strategy to fix the Thread.sleep issue in 
https://github.com/azkaban/azkaban/blob/d0d63361ca9def7190823a754b28b7dc5d9a980c/azkaban-web-server/src/test/java/azkaban/scheduler/QuartzSchedulerTest.java#L133